src/main/scala/piconot/precious/RuleBuilder.scala

```diff
@@ -0,0 +1,85 @@
+package piconot.precious
+
+/*
+ *    When(holding weapon 1)
+      and( Shire is empty).and(LonelyMountain has anything).and.(UndyingLands is empty).and( Mordor has anything)
+      then(go).towards(UndyingLands).and(ready weapon 1),
+
+ */
+
+import picolib.semantics._
+import scala.collection.mutable._
+
+class RuleBuilder() {
+  // Stored in NEWS order
+  var northContents: RelativeDescription = null
+  var eastContents: RelativeDescription = null
+  var westContents: RelativeDescription = null
+  var southContents: RelativeDescription = null
+  var resultDirection: MoveDirection = null
+  var initialState: State = null
+  var finalState: State = null
+  
+  var setInitialState: Boolean = false;
+  var directionToSet: Int = -1;
+  
+  def toRule(): Rule = {
+    val surroundings: Surroundings = Surroundings(northContents,eastContents,westContents,southContents)
+    Rule(initialState, surroundings, resultDirection, finalState)
+  }
+  
+  def you(toHold: Boolean): RuleBuilder = {
+      setInitialState = toHold
+      this
+  } 
+  
+  def and(place: MoveDirection): RuleBuilder = {
+    place match {
+      case North => directionToSet = 0
+      case East => directionToSet = 1
+      case West => directionToSet = 2
+      case South => directionToSet = 3
+    }
+    this
+  }
+  
+  def weapon(weaponNum: State): RuleBuilder = {
+    if (setInitialState) {
+      initialState = weaponNum;
+    } else {
+      finalState = weaponNum;
+    }
+    this
+  }
```

Comment:
i'm not sure I quite understand this, but could you do this with a default argument instead?